File: 9_E.cpp
// 難在將int轉成char

class Solution {
public:
    bool isPalindrome(int x) {
        char const *start;
        char const *end;

        string word = to_string(x);
        start = word.c_str();
        end = start + strlen(start)-1;

        while(start < end){
            if(*start != *end) return false;
            else{
                start++;
                end--;
            }
        }
        return true;
    }
};

File: 9_E.py
# 參考g2g的方法

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        # transfer int to string
        # find reverse of the string
        # check if reverse and original are the same
        word = str(x)
        return word == word[::-1]